Updates docs for smart contracts (#8)
aragon/builders-portal: docs/smart-contracts/core/plugin/PluginTransparentUpgradeable.md

## Aragon Core

###  contract `PluginTransparentUpgradeable`

An abstract contract to inherit from when creating a proxy contract.
This should be used to deploy logic contracts where proxy itself

> is deployed through transparent or beacon...

Reviewer: ```suggestion
is deployed through transparent or beacon.
```

Reviewer: Check for the 3 different lines here again on the UI

#### public variable `PLUGIN_INTERFACE_ID`

```solidity
bytes4 PLUGIN_INTERFACE_ID 
```

#### internal modifier `auth`

```solidity
modifier auth(bytes32 _permissionId) 
```

| Input | Type | Description |
|:----- | ---- | ----------- |
| _permissionId | bytes32 | The hash of the permission identifier |

*Auth modifier used in all components of a DAO to check the permissions.*

#### public function `supportsInterface`

> adds a IERC165 to check whether contract supports PluginTranparentUpgradeable interface or not.

Reviewer: ```suggestion
Checks whether the specified contract supports the PluginTransparentUpgradeable interface (or the IERC165 interface) or not
```

```solidity
function supportsInterface(bytes4 interfaceId) public view virtual returns (bool) 
```

| Output | Type | Description |
| ------ | ---- | ----------- |

> | [0] | bool | bool whether it supports the IERC165 or PluginTranparentUpgradeable |

Reviewer: ```suggestion
| [0] | bool | Boolean answer to whether the contract specified supports the IERC165 or the PluginTranparentUpgradeable interface |
```
